test/utilities.jl

```diff
@@ -0,0 +1,128 @@
+"""
+assert_allclose(actual, reference; [rtol], [atol], [err_msg], [error_location_fmt])
+
+Raises an assertion exception when any entries in `actual` and `reference` are not equal up to the
+desired tolerance. In other words, all entries of `actual` and `reference` must satisfy:
+`abs(actual - reference) <= rtol * abs(reference) + atol`
+
+This is inspired by numpy's assert_allclose function. However, we've changed some default values.
+Additionally, in cases where actual and reference don't have equal values but do have the same 
+shape, the locations of the max errors are also printed.
+
+# Keywords
+- `rtol`: The desired relative tolerance
+- `atol`: The desired absolute tolerance
+- `err_msg`: An optional error message that can be printed when the arrays are not close
+- `error_location_fmt`: An optional keyword used to specify a function that creates a string 
```

Comment:
Are you using these two arguments anywhere in the tests?  They seem a bit overengineered.  Am I missing an important use case?

Reply:
Yeah, this is used by the `assert_allclose_grid` function for pretty printing (see below)